backend/src/core/middlewares/api-key.middleware.ts

```diff
@@ -0,0 +1,50 @@
+import { loggerWithLabel } from '@core/logger'
+import { Handler, NextFunction, Request, Response } from 'express'
+import { ApiKeyService, AuthService } from '@core/services'
+
+export interface ApiKeyMiddleware {
+  deleteApiKey: Handler
+}
+export const InitApiKeyMiddleware = (authService: AuthService) => {
+  const logger = loggerWithLabel(module)
+
+  const deleteApiKey = async (
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ): Promise<Response | void> => {
+    try {
+      // enforce authentication by cookie
+      const authenticated = authService.checkCookie(req)
+      if (!authenticated) {
+        return res.sendStatus(403).json({
+          message: `Invalid cookie`,
+        })
```

Comment:
**[change request]** this part is a bit weird, cmiiw but shouldn't authentication be handled at the route level (i.e. in `backend/src/core/routes/index.ts`), rather than here? 

reasons for this:
- at the route level, you can clearly distinguish which routes are meant to accept which modes of validation (so in this case, we only accept cookie)
- clearer separation of concerns

Reply:
good point, changing next push